src/test/ui/suggestions/match-ergonomics.stderr

```diff
@@ -0,0 +1,53 @@
+error[E0308]: mismatched types
+  --> $DIR/match-ergonomics.rs:4:10
+   |
+LL |         [&v] => {},
+   |          ^^
+   |          |
+   |          expected i32, found reference
+   |          help: you can probably remove the explicit borrow: `v`
+   |
+   = note: expected type `i32`
+              found type `&_`
+
+error[E0529]: expected an array or slice, found `std::vec::Vec<i32>`
+  --> $DIR/match-ergonomics.rs:8:9
+   |
+LL |         [&v] => {},
+   |         ^^^^ pattern cannot match with input type `std::vec::Vec<i32>`
+
+error[E0529]: expected an array or slice, found `std::vec::Vec<i32>`
+  --> $DIR/match-ergonomics.rs:20:9
+   |
+LL |         [v] => {},
+   |         ^^^ pattern cannot match with input type `std::vec::Vec<i32>`
+
+error[E0308]: mismatched types
+  --> $DIR/match-ergonomics.rs:29:9
+   |
+LL |         &v => {},
+   |         ^^ expected i32, found reference
+   |
+   = note: expected type `i32`
+              found type `&_`
+help: you can rely on match ergonomics and remove the explicit borrow
```

Comment:
this hint seems odd to me. `y` is an `i32`, so that match will not be doing any match ergonomics. This could just have the same suggestion as the other places with too many references ("you can probably remove the explicit borrow")

Reply:
Done